src/components/ProfileCard.tsx

```diff
@@ -0,0 +1,67 @@
+import {User} from "next-auth";
+import {api} from "~/utils/api";
+import {timeAgo} from "~/utils/time";
+import {ProfilePicture} from "./ProfilePicture";
+import {Spinner} from "./Spinner";
+
+interface ProfileCardProps {
+    dateJoined?: Date,
+    name: string
+    imgUrl?: string | null,
+    userId: string,
+}
+
+export const ProfileCard: React.FC<ProfileCardProps> = ({
+    dateJoined,
+    name,
+    imgUrl,
+    userId,
+}) => {
+    const { data: nBlogPosts } = api.blogPost.count.useQuery({
+        userId: userId 
+    });
+    const { data: nComments } = api.comment.count.useQuery({
+        userId: userId
+    });
+
+    let isLoading = false;
+    if (nComments == undefined || nBlogPosts == undefined) { 
+        isLoading = true;
+    }
```

Comment:
These should go in a `useEffect`. `useEffect` executes a code block when any the value of any variables in the dependency array change. Combined with the `useState` comment above would look something like this:

```jsx
useEffect(() => {
  setIsLoading(true);
}, [nComments, nBlogPosts])
```

Reply:
took Altay's suggestion